jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/assembly/JKubeBuildTarArchiver.java

```diff
@@ -81,4 +79,8 @@ public File createArchive(File inputDirectory, BuildDirs buildDirs, ArchiveCompr
         return JKubeTarArchiver.createTarBall(outputFile, inputDirectory, fileListToAddInTarball, fileModeMap, compression);
     }
 
+    private boolean shouldCopyFileToAssemblyDir(File srcFile, File targetFile) throws IOException {
+        return !srcFile.getCanonicalPath().equals(targetFile.getCanonicalPath());
```

Comment:
I don't understand the purpose of this statement.

Is this supposed to prevent copying a file to "itself"? if this is the case, is it possible to at least change to something like `!srcFile.equals(targetFile)` to make it more obvious?

However, I'm still not sure how this belongs here. It should make more sense within the `FileUtils#copyFile` method.